src/app/api/view-settings/viewSettings.controller.ts

```diff
@@ -0,0 +1,24 @@
+import { NextRequest, NextResponse } from 'next/server'
+import authenticate from '@api/core/utils/authenticate'
+import { ViewSettingsService } from '@api/view-settings/viewSettings.service'
+import { CreateViewSettingsSchema } from '@/types/dto/viewSettings.dto'
+
+export const getViewSetting = async (req: NextRequest) => {
+  const user = await authenticate(req)
+
+  const viewSettingsService = new ViewSettingsService(user)
+  const viewSetting = await viewSettingsService.getViewSettingsForUser()
+
+  return NextResponse.json({ viewSetting })
+}
+
+export const updateViewSetting = async (req: NextRequest) => {
+  const user = await authenticate(req)
+
+  const data = CreateViewSettingsSchema.parse(await req.json())
+
+  const viewSettingsService = new ViewSettingsService(user)
+  const newViewSetting = await viewSettingsService.createOrUpdateViewSettings(data)
+
+  return NextResponse.json({ newViewSetting })
```

Comment:
why do we pass this in using a `newViewSettings` object? IMO we can just pass in the object representing the view settings directly as the body.

Reply:
Answered: https://github.com/copilot-platforms/tasks-app/pull/20#discussion_r1583197394